README.md

```diff
@@ -10,32 +10,275 @@
+**:information_source: Notes about the command format:**<br>
 
-* Create a new module
-* Format: `add module m/<module name>`
+* Lowercase letter(s) followed by a slash are prefixes that prefix a parameter, to be supplied by the user.<br>
+  e.g. in `add module m/<module name>`, `m/` is a prefix that prefixes the `<module name>` parameter.
 
-#### Module add task
+* Words in `<angle brackets>` are the parameters prefixed by prefixes, to be supplied by the user.<br>
+  e.g. in `add module m/<module name>`, `<module name>` is a parameter which can be used as `add module m/CS2105`.
 
-* Create a new task for a specific module
-* Format: `add task m/<module name> a/<task name> d/<task deadline>`
+* Parameters must be prefixed with their specific prefixes.<br>
+  e.g. the command `add module m/CS2105` will add the module CS2105 to TAB, where the commands `add module i/CS2105` and `add module CS2105` will not work.
 
-#### Module add student
+* Parameters must only contain permitted characters, as listed in the [Parameter summary](#parameter-summary-with-prefix-included) below.
 
-* Add a student to a specific module
-* Format: `add student m/<module name> i/<student id> n/<student name> t/<tele_handle> e/<student email>`
+* Optional parameters need not be all present when executing a command, they are denoted in this user guide using square brackets '[]'.<br>
+  e.g. in `edit task m/<module name> ti/<task id> [a/<edited task name>] [d/<edited task deadline>]`, `a/<edited task name>` and `d/<edited task deadline>` are the optional parameters
 
-#### Module delete student
+</div>
 
-* Delete a student from a specific module
-* Format: `module delete <module name>, <student id>`
 
-#### Module delete
+### Add a new module: `add module`
 
-* Delete a module
-* Format: `module delete <module name>`
+Add a new module to TAB.<br>
+Module name must be an NUS module code, consisting of 2-4 capital letters, followed by 4 numbers, followed by 0-4 capital letters.
+Format: `add module m/<module name>`
+
+Examples:
+* `add module m/CS2103`: Add module CS2103 to TAB
+* `add module m/CS2100`: Add module CS2100 to TAB
+
+### Edit a module's name: `edit module`
+
+Edit the name of the module.<br>
+Module name must be an NUS module code, consisting of 2-4 capital letters, followed by 4 numbers, followed by 0-4 capital letters.
+Format: `edit module m/<old module name> mn/<new module name>`
+
+Examples:
+* `edit module m/CS2103 mn/CS2105`:
+    * Edit the module name of module "CS2103" to "CS2105"
+
+### Delete a module from TAB: `delete module`
+
+Delete the specified module from TAB.
+Module name must be an NUS module code, consisting of 2-4 capital letters, followed by 4 numbers, followed by 0-4 capital letters.
+Format: `delete module m/<module name>`
+
+Examples:
+* `delete module m/CS2103`:
+    * Delete module CS2103 from TAB
+* `delete module m/CS2100`:
+    * Delete module CS2100 from TAB
+
+### Add a student to a specific module: `add student`
```

Comment:
Maybe add "if the student is already present in other modules, the student must have the exact identities for him/her to be successfully added into the module"

Reply:
agreed